builtin/docker/platform.go

```diff
@@ -141,43 +144,80 @@ func (p *Platform) Status(
 
 	log.Debug("querying docker for container health")
 
-	resources := []*sdk.StatusReport_Resource{{
-		Name: containerInfo.Name,
-	}}
+	// Find the declared resource that corresponds to our observed container
+	var containerDeclaredResource *sdk.DeclaredResource
```

Comment:
Leaving the work of mapping resources to declared resources to the plugin devs this way feels a bit clunky to me, but I think we can make this easier with ResourceManager.